src/common/GasService.sol

```diff
@@ -46,29 +46,31 @@ contract GasService is IGasService {
 
         // NOTE: The hardcoded values are take from the EndToEnd tests. This should be automated in the future.
 
-        scheduleUpgrade = BASE_COST + SMALL_COST;
-        cancelUpgrade = BASE_COST + SMALL_COST;
-        recoverTokens = BASE_COST + SMALL_COST;
+        scheduleUpgrade = BASE_COST + 28514;
+        cancelUpgrade = BASE_COST + 8861;
+        recoverTokens = BASE_COST + 82906;
         registerAsset = BASE_COST + 34329;
         request = BASE_COST + 86084; // request deposit case
         notifyPool = BASE_COST + 38190;
         notifyShareClass = BASE_COST + 1775916;
         notifyPricePoolPerShare = BASE_COST + 30496;
         notifyPricePoolPerAsset = BASE_COST + 35759;
-        notifyShareMetadata = BASE_COST + SMALL_COST;
-        updateShareHook = BASE_COST + SMALL_COST;
+        notifyShareMetadata = BASE_COST + 13343;
+        updateShareHook = BASE_COST + 6415;
         initiateTransferShares = BASE_COST + 52195;
         executeTransferShares = BASE_COST + 70267;
         updateRestriction = BASE_COST + 35992;
-        updateContract = BASE_COST + SMALL_COST;
+        updateContract = BASE_COST + 53345;
         requestCallback = BASE_COST + 186947; // approve deposit case
-        updateVault = BASE_COST + 2770342; // deploy vault case
+        updateVaultDeployAndLink = BASE_COST + 2770342;
+        updateVaultLink = BASE_COST + 100567;
+        updateVaultUnlink = BASE_COST + 20814;
```

Comment:
Note `DeployAndLink` is 130 times higher than `Unlink` so I think this optimization makes sense for this message